src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -753,6 +805,34 @@ def _get_node_referenced_flow_inputs(
                 node_referenced_flow_inputs[value.value] = flow_inputs[value.value]
         return node_referenced_flow_inputs
 
+    def _exec_prepare(self, inputs, run_id, line_number, variant_id):
```

Comment:
The function returns a 4-elements tuple, and it seems very hard to think of a proper function name, maybe we can keep duplicated code for now.

@thy09 what do you think?

Reply:
Can we use `with` to achieve this?
This ask is not that hurry so we can take more time to improve these.

Reply:
Another point, since variant_id is no longer used, we can remove this param.